### DISCUSSION TITLE
Chop Shop

### DISCUSSION THREAD

**Discussion by ModularAU** [EveronLife/EveronLife#147] [Ideas]
A Chop shop, to sell stolen Vehicles at

**Current Situation**

Most Life servers have some way of selling stolen cars. I think this would  tie in perfectly with the license plate system, as it could force some cool RP situations with Police etc.

**New System**

A Unique trader for selling stolen vehicles

I would assume we would need to add some way of entering stolen vehicles, Perhaps this can be something like a lockpick, crowbar or slim-jim. It adds a lot of use to the License plate system, as police can do searches on cars marked as stolen etc.

I'm not sure weather to make the added tool a item that can be used xyz times, or perhaps a % based system, and if the theft is  unsuccessful, the item is removed.

Again, with the vehicles, I'm not sure weather its worth making the cars sold a % of the MSRP or a flat rate

**Configuration**

Unique tool to enter locked vehicles, with a limited use or % of breaking system

Configurable price for vehicles at chop shop (% based or Flat rate)

Keen to see what everyone thinks!

**Comment by 12gabriel3** [comment]
I've been thinking a lot about the dynamic economy lately and I'll share my perspective coming from this "background".

In another discussion I suggested a system where players would produce "car parts". Since it is a dynamic economy, there should be demand from this item as well. I proposed that each car should take "car parts" from a pool in the market where it is located. If cars aren't semi-disposable, there would never have a demand for car parts, unless they required maintenance, which I think should also consume car parts.

Stolen vehicles are a great _sink_ for car parts. I think when a player sells the stolen car, it should not be converted back into car parts (maybe the buyer export the cars from a port?) and should be taken out of the market. The player should need to buy another vehicle again, increasing the demand for car parts.

This would have a great impact on the economy though. I think this system would work best if we had different _classes_ of vehicles, each specialized for different scenarios and we should take that into account when balancing, and the rule for players should be: "Don't drive what you cannot lose"

Firstly, I think we should have "safe zones". I think having invincible NPCs guarding the area that shoot anyone holding a weapon would be great in an immersion perspective. But we could also achieve that by different manners. It doesn't really matter, but it should make robbing vehicles in that area nearly impossible.

The _classes_ of vehicles we could have are:

* Beginner vehicles: Slow and moves little cargo, but its basically free. An all-rounded vehicle, not good at anything, but does the job. Could be the UAZ
* Bulk transport: Slow but moves a lot of cargo. Very expensive, should be only used on low risk zones because of speed. Could be the 2 trucks.
* "Sport": little cargo but very fast. Expensive. Great for illegal runs that do not take much room. Could be the M151A2
* Fast transport: Medium cargo and medium speed. The best for transporting illegal goods. Expensive. Could be the HUMVEE

This allows players who got their vehicle stolen to buy a basic transport vehicle while creating demand for car parts by taking the vehicle off the market completely.

We have to be careful to make having your car stolen preferable to dying to avoid a KOS mentality.

This leads to armed robbery. If giving up your vehicle is preferable to death, I think we should implement a robust safety system.

The safety system could have 2 modes, safety on, or safety off. 

* Safety off: the players can _initiate_ combat with others. They do not _comply_ to initiation automatically. Not complying to an _initiation_ gives the other player the right to kill you.
* Safety on: player cannot initiate and complies to _initiations_ automatically. Meaning they stop the vehicle, leave and put their hands up. This does not give a kill right to the other player. Killing a player without a kill right should be either impossible or have bad consequences.

Having these rules integrated with gameplay mechanics allows the server to have less rules, less bans being issued, frees up admin work and so on.

Having a safety system instead of a timer to comply allows for more ambushes and avoids situations where the car flies past you when the timer is still running, giving you no chance to fire at them. We could decrease the timer, but the safety system would avoid human error, such as mis clicking or not complying by accident.

**Reply by ModularAU** [reply]
Heya mate! I saw your post before in #146 that mentioned car parts to be honest, and this gave me the chop shop idea.

I definitely think there needs to be some sort of balancing, to not entice KOS and constant rebel RP as you've mentioned.

- Weather this is a cooldown on selling said stolen cars, or if its PDE, the more stolen cars that are sold, the price is lower / ports become too heated if too many are sold in a short time, so you need to wait for them to cool down.

- Or perhaps a random chance for the police to be alerted etc.


I agree on if your car is chopped you cannot get it back, (OR can but have to buy it from elsewhere? perhaps there is a dealership  of sorts for illegal cars that have been stolen that are cheaper, but run a risk of driving etc.) this ties back in to the PDE a

I think it would be awesome as it could create  some sort of system for running fake number plates as well? I having someone be a prat to you at a safezone, and clocking their plate, and then faking that in a robbery. or some fast and furious type escape with 20 cars and all the same number plates lol.

Weather this is at the chop shop, who sell you custom plates for a fee, or else where.

Then I guess, you can also make cars identifiable via VIN, so although plates are a replica, the original plates are uniquely tied to the car via 
VIN, again, tying into a feedback loop with forcing RP with police etc, as if they have suspicion plates are stolen, youd need to be pulled over and verified etc.

Keen to see what anyone else thinks!

**Reply by thebonbon** [reply]
In Arma 3 you just had a random car just to move your resources from a to b. If you left it somewhere for 2 seconds it was stolen and sold at the chop shop. 
I hated this feature that you could loose your car forever.

In GTA you could change color, upgrade it and even change the honk sound. It was **your** car and not something random that if it blows up or stolen you just buy a new one. I really like if you see a cool / unique sports car and be like "oh thats xyz´s car".

Imo we should not add a permanent deletion feature for cars. Only thing is the police can impound it until you pay for it to be released.
..

I do like the lockpick feature and if you fail the items gets deleted.


> Firstly, I think we should have "safe zones". I think having invincible NPCs guarding the area that shoot anyone holding a weapon would be great in an immersion perspective.
>
I think this would be more of a Server decision and not framework. Also invincible NPCs that KOS yeah sure immersion but not for RP imo.

> ..it should make robbing vehicles in that area nearly impossible.
>
Again "no robbing areas" should be a Server decision. Same for the safety system.

**Reply by ModularAU** [reply]
Defo agree on gta making your car feel like your car! I’m interested to see if it’s possible to replicate that in arma though, and how we can do so instead of haha model change go brrr with a colour picker

**Reply by thebonbon** [reply]
It is surprisingly easy to make a tuning system. Vehicles already have a slot system (Ural Tank for example). The tank on the back is just a slot that you can enable / disable or change.
We can easily apply this to every part of a vehicle.
![image](https://user-images.githubusercontent.com/45827986/182158460-fb075fdd-ff9b-41ee-9b16-857d368f9e5d.png)
